### ISSUE TITLE
Deprecation Errors in PHP 8.1.1

### ISSUE THREAD

**Issue by secondmanveran** [dauxio/daux.io#292]
**Describe the bug**
Running the `generate` command produces deprecation errors running in the terminal on macOS Monterey  12.1 and PHP v8.1.1

**To Reproduce**
1. Install the package globally via composer
2. Navigate to dir containing `docs` dir
3. Run `daux generate` or simply `daux`

**Expected behavior**
Run without deprecation errors

**Screenshots**
```
Deprecated: Return type of Todaymade\Daux\Tree\Directory::offsetGet($offset) should either be compatible 
with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used 
to temporarily suppress the notice in 
/Users/secondman/.composer/vendor/daux/daux.io/libs/Tree/Directory.php on line 282

Deprecated: Return type of Todaymade\Daux\Tree\Directory::offsetSet($offset, $value) should either be compatible 
with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used 
to temporarily suppress the notice in 
/Users/secondman/.composer/vendor/daux/daux.io/libs/Tree/Directory.php on line 293

Deprecated: Return type of Todaymade\Daux\Tree\Directory::offsetUnset($offset) should either be compatible 
with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used 
to temporarily suppress the notice in 
/Users/secondman/.composer/vendor/daux/daux.io/libs/Tree/Directory.php on line 307

Deprecated: Return type of Todaymade\Daux\Tree\Directory::getIterator() should either be compatible 
with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used 
to temporarily suppress the notice in 
/Users/secondman/.composer/vendor/daux/daux.io/libs/Tree/Directory.php on line 312
```

**Desktop (please complete the following information):**
 - OS: macOS Monterey
 - Browser N/A terminal errors
 - Version 12.1
 - PHP Version: 8.1.1

**Smartphone (please complete the following information):**
 N/A -- terminal errors

**Additional context**
N/A

**Comment by onigoetz** [comment]
Hi, the warnings are now fixed in master

**Comment by secondmanveran** [comment]
@onigoetz 

Great thanks. Any ETA on a release I can pull down?

**Comment by onigoetz** [comment]
I have another bug to check out before I can make a release, should be released by monday

**Comment by secondmanveran** [comment]
Brilliant

**Comment by onigoetz** [comment]
Released in version [0.18.0](https://github.com/dauxio/daux.io/releases/tag/0.18.0)
